### ISSUE TITLE
Nvim support

### ISSUE THREAD

**Issue by Snxwman** [bootdotdev/curriculum#31]
Thank you for implementing vim mode in the online editor!
 
Going further, it would be really nice to be able to work in our local nvim install rather than having to use the website. A format similar to the [leetcode plugin](https://github.com/kawre/leetcode.nvim) could be interesting.

**Comment by jmiguelhdez** [comment]
i agree with the sentiment on wanting to use our own editors for the courses, however, i feel like a cli app would be nice instead of focusing on one editor in particular. That way we could make plugins ourselves for any editor we want.

**Comment by MrDwarf7** [comment]
> i agree with the sentiment on wanting to use our own editors for the courses, however, i feel like a cli app would be nice instead of focusing on one editor in particular. That way we could make plugins ourselves for any editor we want.

Perhaps they will have us build our own application to from scratch in Rust or C++ that we can then use for the rest of the course?
(Teej??? Prime's return to Rust? Zig...?)
